src/Metal/MTLRenderCommandEncoder.cs

```diff
@@ -10,29 +11,45 @@
 
 namespace Metal {
 	public static class IMTLRenderCommandEncoder_Extensions {
+#if !NET
 		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#else
+		[SupportedOSPlatform ("macos10.13")]
+#endif
 		public unsafe static void SetViewports (this IMTLRenderCommandEncoder This, MTLViewport [] viewports)
 		{
 			fixed (void* handle = viewports)
 				This.SetViewports ((IntPtr)handle, (nuint)(viewports?.Length ?? 0));
 		}
 
+#if !NET
 		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#else
+		[SupportedOSPlatform ("macos10.13")]
+#endif
 		public unsafe static void SetScissorRects (this IMTLRenderCommandEncoder This, MTLScissorRect [] scissorRects)
 		{
 			fixed (void* handle = scissorRects)
 				This.SetScissorRects ((IntPtr)handle, (nuint)(scissorRects?.Length ?? 0));
 		}
 
 #if IOS
+#if !NET
 		[iOS (11,0), NoTV, NoMac, NoWatch]
+#else
+		[SupportedOSPlatform ("ios11.0")]
+#endif
 		public unsafe static void SetTileBuffers (this IMTLRenderCommandEncoder This, IMTLBuffer[] buffers, nuint[] offsets, NSRange range)
 		{
 			fixed (void* handle = offsets)
 				This.SetTileBuffers (buffers, (IntPtr)handle, range);
 		}
 
+#if !NET
 		[iOS (11,0), NoTV, NoMac, NoWatch]
+#else
+		[SupportedOSPlatform ("ios11.0")]
+#endif
```

Comment:
uho, that file seems to be under a `#if MONOMAC` so this is not exposed

Reply:
@tj-devel709  we should fix this.

Reply:
Sorry, there is another thread below about the same issue.
There is an issue created here to fix in another PR #12648